src/client/Microsoft.Identity.Client/ManagedIdentity/V2/IHybridCache.cs

```diff
@@ -0,0 +1,120 @@
+        /// The cache key used to identify the token entry. Should be the same key used in <see cref="GetAsync"/>.
+        /// </param>
+        /// <param name="token">
+        /// The attestation token to cache. This is typically a JWT (JSON Web Token) in string format.
+        /// Must not be null or empty.
+        /// </param>
+        /// <param name="expiresOnUtc">
+        /// The UTC date and time when the token expires. The cache implementation may add additional
+        /// buffer time to account for clock skew and ensure tokens are refreshed before actual expiration.
+        /// </param>
+        /// <param name="ct">
+        /// A cancellation token that can be used to cancel the asynchronous operation.
+        /// </param>
+        /// <returns>
+        /// A task that represents the asynchronous set operation.
+        /// </returns>
+        /// <remarks>
+        /// This method should:
+        /// - Overwrite existing entries for the same key
+        /// - Handle storage errors gracefully (cache failures should not fail the authentication flow)
+        /// - Be thread-safe for concurrent access
+        /// - Validate input parameters
+        /// - For hybrid implementations: update both in-memory and persistent storage when possible
+        /// - Prioritize fast in-memory updates over slower persistent operations
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when <paramref name="token"/> is null or empty.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">
+        /// Thrown when the operation is canceled via the cancellation token.
+        /// </exception>
+        Task SetAsync(long key, string token, DateTimeOffset expiresOnUtc, CancellationToken ct);
+
+        /// <summary>
+        /// Removes an expired or invalid token from the cache.
+        /// </summary>
+        /// <param name="key">
+        /// The cache key identifying the token entry to remove.
+        /// </param>
+        /// <param name="ct">
+        /// A cancellation token that can be used to cancel the asynchronous operation.
+        /// </param>
+        /// <returns>
+        /// A task that represents the asynchronous remove operation.
+        /// </returns>
+        /// <remarks>
+        /// This method should:
+        /// - Gracefully handle cases where the key doesn't exist (no-op)
+        /// - Handle storage errors gracefully
+        /// - Be thread-safe for concurrent access
+        /// - Not throw exceptions for missing entries
+        /// - For hybrid implementations: remove from both in-memory and persistent storage
+        /// - Continue operation even if one storage tier fails
+        /// </remarks>
+        /// <exception cref="OperationCanceledException">
+        /// Thrown when the operation is canceled via the cancellation token.
+        /// </exception>
+        Task RemoveAsync(long key, CancellationToken ct);
```

Comment:
nit: I believe it's customary to return Task<bool> to indicate if an entry was actually removed.